### PR TITLE
Copter: disable landing gear on v2 build

### PR DESCRIPTION
**Pull request by tatsuy**
I defined LANDING_GEAR_ENABLED, and then I set HAL_MINIMIZE_FEATURES to it same as Plane.
https://github.com/ArduPilot/ardupilot/blob/master/ArduPlane/config.h#L302

This option saves 1.7k of flash memory.
I also add option to disable LANDING_GEAR.

- Before
```
Build directory: /home/drone/github/ardupilot/build/fmuv2
Target          Text    Data  BSS     Total  
---------------------------------------------
bin/arducopter  953424   832  195984  1150240
```

- After
```
Build directory: /home/drone/github/ardupilot/build/fmuv2
Target          Text    Data  BSS     Total  
---------------------------------------------
bin/arducopter  951644   832  195988  1148464
```

### CODE REVIEW

**Review by khancyr — APPROVED** [review]
LGTM

**Comment by rmackay9** [comment]
Being consistent with plane makes some sense but at the same time I wonder why we would do this.  Are we running out of space in the v2 builds?

File: ArduCopter/config.h [hunk 1]
```diff
@@ -377,6 +377,12 @@
  # define BUTTON_ENABLED ENABLED
 #endif
 
+//////////////////////////////////////////////////////////////////////////////
+// Landing Gear support
+#ifndef LANDING_GEAR_ENABLED
+ #define LANDING_GEAR_ENABLED !HAL_MINIMIZE_FEATURES
```
Comment (tatsuy) [inline ArduCopter/config.h:383]:
```suggestion
 #define LANDING_GEAR_ENABLED ENABLED
```

**Review by tatsuy — COMMENTED** [review]
I noticed that it may be a risk to change it.
If there are existing users, the change will make them confuse.

I can keep it ENABLED.
We can judge to disable it if we run out of space in the future.

My main purpose is to add the option to disable LANDING_GEAR.

**Comment by khancyr** [comment]
hello @tatsuy,

Could you rebase this, we should be able to merge it now as it won't change the default behavior !

**Review by rmackay9 — APPROVED** [review]

**Comment by rmackay9** [comment]
The sub test failed so I've kicked it off again.

**Comment by rmackay9** [comment]
Merged, thanks!

**Comment by tatsuy** [comment]
Thank you!

**Pull request merged**